File: QuickBooksAPI/DataAccessLayer/Models/Vendor.cs
namespace QuickBooksAPI.DataAccessLayer.Models
{
    public class Vendor
    {
        public int Id { get; set; } // Local PK
        public string QboId { get; set; }
        public string UserId { get; set; }
        public string RealmId { get; set; }
        public string SyncToken { get; set; }
        public string Title { get; set; }
        public string GivenName { get; set; }
        public string MiddleName { get; set; }
        public string FamilyName { get; set; }
        public string DisplayName { get; set; }
        public string CompanyName { get; set; }
        public bool Active { get; set; }
        public decimal Balance { get; set; }
        public string PrimaryEmailAddr { get; set; }
        public string PrimaryPhone { get; set; }
        public string BillAddrLine1 { get; set; }
        public string BillAddrCity { get; set; }
        public string BillAddrPostalCode { get; set; }
        public string BillAddrCountrySubDivisionCode { get; set; }
        public DateTime CreateTime { get; set; }
        public DateTime LastUpdatedTime { get; set; }
        public string Domain { get; set; }
        public bool Sparse { get; set; }
        /// <summary>Set when vendor is soft-deleted; null means active.</summary>
        public DateTime? DeletedAt { get; set; }
        /// <summary>UserId who performed the soft delete.</summary>
        public string? DeletedBy { get; set; }
    }
}

File: QuickBooksAPI Frontend/app/src/types/vendor.ts
export interface Vendor {
  id: number;
  qboId: string;
  userId: string;
  realmId: string;
  syncToken: string;
  title: string;
  givenName: string;
  middleName: string;
  familyName: string;
  suffix: string;
  displayName: string;
  companyName: string;
  printOnCheckName: string;
  active: boolean;
  balance: number;
  primaryEmailAddr: string;
  primaryPhone: string;
  mobile: string;
  webAddr: string;
  taxIdentifier: string;
  acctNum: string;
  billAddrLine1: string;
  billAddrLine2: string;
  billAddrLine3: string;
  billAddrCity: string;
  billAddrPostalCode: string;
  billAddrCountrySubDivisionCode: string;
  billAddrCountry: string;
  domain: string;
  sparse: boolean;
  createTime: string;
  lastUpdatedTime: string;
  deletedAt: string | null;
  deletedBy: string | null;
}

export interface CreateVendorRequest {
  displayName: string;
  givenName?: string;
  middleName?: string;
  familyName?: string;
  companyName?: string;
  title?: string;
  suffix?: string;
  primaryEmailAddr?: { address: string };
  primaryPhone?: { freeFormNumber: string };
  mobile?: { freeFormNumber: string };
  webAddr?: { uri: string };
  billAddr?: {
    line1: string;
    line2?: string;
    line3?: string;
    city: string;
    postalCode: string;
    countrySubDivisionCode: string;
    country?: string;
  };
  printOnCheckName?: string;
  acctNum?: string;
  taxIdentifier?: string;
}

export interface UpdateVendorRequest {
  id: string;
  syncToken: string;
  sparse?: boolean;
  displayName?: string;
  givenName?: string;
  middleName?: string;
  familyName?: string;
  companyName?: string;
  title?: string;
  suffix?: string;
  primaryEmailAddr?: { address: string };
  primaryPhone?: { freeFormNumber: string };
  mobile?: { freeFormNumber: string };
  webAddr?: { uri: string };
  billAddr?: {
    line1: string;
    line2?: string;
    line3?: string;
    city: string;
    postalCode: string;
    countrySubDivisionCode: string;
    country?: string;
  };
  printOnCheckName?: string;
  acctNum?: string;
  taxIdentifier?: string;
  active?: boolean;
  balance?: number;
}

export interface SoftDeleteVendorRequest {
  id: string;
  syncToken: string;
}
